### DISCUSSION TITLE
Unit testing targeted OS/device configurations

### DISCUSSION THREAD

**Discussion by GeorgeLeithead** [CommunityToolkit/Maui#615] [General]
Hi,

Is it possible in `CommunityToolkit.Maui.UnitTests` to have unit tests target specific platforms?  To have valid tests where specific platforms are indicated, such as (not great examples, but...):

```csharp
#if WINDOWS
	this.HandlerChanged += OnPopupHandlerChanged;
#endif
```

and

```csharp
// Work-around for iOS / MacOS bug that paints `Border.BackgroundColor` over top of `Border.Content` when an Image is used for `Border.Content`
if (OperatingSystem.IsIOS() || OperatingSystem.IsMacCatalyst() || OperatingSystem.IsMacOS())
{
	BackgroundColor = Colors.Transparent;
}
```
Thanks, George.

**Comment by TheCodeTraveler** [comment]
Right now, `CommunityToolkit.Maui.UnitTest.csproj` only targets `net6.0`. I'm unsure if Unit Tests can run on `net6.0-android` and `net6.0-ios`. You're welcome to try adding these target frameworks and see if it works!

**Reply by pictos** [reply]
@GeorgeLeithead if you want, you can start the quest from [this branch](https://github.com/CommunityToolkit/Maui/tree/pj/device-test), there was some limitations at the time that I tried, but maybe they're fixed now... Also, Allan R. released a runner version that he extracted from .NET MAUI, maybe that fits better. Let me know if you need anything

**Reply by GeorgeLeithead** [reply]
@pictos  As soon as I get all the other things that I'm working on for MCT in a good state, I'll have a look at branching that branch. @bijington did say he thought you were working on something like this.

**Comment by CliffAgius** [comment]
Closed as answered
